### ISSUE TITLE
Parallel tests fail to run mockk's coEvery in beforeTest lifecycle

### ISSUE THREAD

**Issue by sharmando** [kotest/kotest#3117]
I am attempting to run multiple Specs in parallel. After increasing parallelism some of my test fail.

The reason of the failure is that the coEvery mockking functionality hasn't run and mockk fails with missing implementation.

I am not sure if this is a bug in kotest (lifecycle and parallelism) or I am just using it wrong.

* kotest 5.3.2
* kotlin 1.6.21
* coroutines 1.6.4
* mockk 1.12.4

**Comment by sksamuel** [comment]
MockK often has problems with race conditions. We've seen this in the past.

**Comment by sharmando** [comment]
Hi Sam, thanks for the quick response. 

Is there mocking library known to work with kotest in parallel? Mockito for instance? I am trying to run parallel mocking tests.

Should we make a note somewhere in the docs to say that mockk works with kotest but fails to do so when parallelism is on?

**Comment by LeoColman** [comment]
Personally, I'd say that if you're mocking too many things that are prone to race conditions, Mockk or any other mocking framework will be very delicate around it. My guess is even Mockito.

Have you tried to play with Isolation Mode? https://kotest.io/docs/framework/isolation-mode.html

When racing with mocks, `InstancePerTest` usually saves me

**Comment by sharmando** [comment]
@LeoColman, thanks for the recommendations they make total sense. However, I have successfully run parallel mocking tests before (with other libs and language) and I made sure the code I was testing was side effect free.

My problem was to assume that clearAllMocks would somehow figure out what are the mocks in the current Spec, it knows nothing and it was clearing mocks across all tests running in parallel. (Naturally, Isolation Modes also did not help with that issue)

Happy to close this and to know that I can run parallel tests using mockk and kotest. I did read something about static mocks not being thread safe in mockk but I don't have plans to use that.
